homeassistant/components/dwd_weather_warnings/sensor.py

```diff
@@ -109,55 +108,65 @@ async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up entities from config entry."""
-    api = WrappedDwDWWAPI(hass.data[DOMAIN][entry.entry_id])
+    api = hass.data[DOMAIN][entry.entry_id]
+    coordinator = DwdWeatherWarningsCoordinator(hass, api)
+
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
This should go in `__init__.py` as `async_config_entry_first_refresh` needs to raise there if anything is wrong with the first update.
Secondly you should store the coordinator at `hass.data[DOMAIN][entry.entry_id]` and not `api` as `api` is now part of the coordinator so it's already there